test/e2e/broker_default_test.go

```diff
@@ -43,135 +42,229 @@ const (
+			Data:           fmt.Sprintf(`{"msg":%q}`, body),
+		}
+	} else {
+		return &resources.CloudEvent{
 			Source: eventToSend.Source,
 			Type:   eventToSend.Type,
 			Data:   fmt.Sprintf(`{"msg":%q}`, body),
 		}
-		// Create sender pod.
-		senderPodName := name("sender", eventToSend.Type, eventToSend.Source)
-		if err := client.SendFakeEventToAddressable(senderPodName, defaultBrokerName, common.BrokerTypeMeta, cloudEvent); err != nil {
-			t.Fatalf("Error send cloud event to broker: %v", err)
-		}
-
-		// Check on every dumper whether we should expect this event or not, and add its body
-		// to the expectedEvents/unexpectedEvents maps.
-		for _, eventToReceive := range eventsToReceive {
-			subscriberName := name("dumper", eventToReceive.typeAndSource.Type, eventToReceive.typeAndSource.Source)
-			if shouldExpectEvent(&eventToSend, &eventToReceive, t.Logf) {
-				expectedEvents[subscriberName] = append(expectedEvents[subscriberName], body)
-			} else {
-				unexpectedEvents[subscriberName] = append(unexpectedEvents[subscriberName], body)
-			}
-		}
 	}
+}
 
-	for _, event := range eventsToReceive {
-		subscriberName := name("dumper", event.typeAndSource.Type, event.typeAndSource.Source)
-		if err := client.CheckLog(subscriberName, common.CheckerContainsAll(expectedEvents[subscriberName])); err != nil {
-			t.Fatalf("Event(s) not found in logs of subscriber pod %q: %v", subscriberName, err)
-		}
-		// At this point all the events should have been received in the pod.
-		// We check whether we find unexpected events. If so, then we fail.
-		found, err := client.FindAnyLogContents(subscriberName, unexpectedEvents[subscriberName])
-		if err != nil {
-			t.Fatalf("Failed querying to find log contents in pod %q: %v", subscriberName, err)
-		}
-		if found {
-			t.Fatalf("Unexpected event(s) found in logs of subscriber pod %q", subscriberName)
+func getTriggerFilterOption(deprecatedTriggerFilter, extensionExists bool, eventMeta eventMeta) resources.TriggerOption {
+	if deprecatedTriggerFilter {
+		return resources.WithDeprecatedSourceAndTypeTriggerFilter(eventMeta.Source, eventMeta.Type)
+	} else {
+		if extensionExists {
+			return resources.WithAttributesAndExtensionTriggerFilter(eventMeta.Source, eventMeta.Type, eventMeta.ExtensionName, eventMeta.ExtensionValue)
+		} else {
+			return resources.WithAttributesTriggerFilter(eventMeta.Source, eventMeta.Type)
 		}
 	}
 }
 
 // Helper function to create names for different objects (e.g., triggers, services, etc.).
-func name(obj, eventType, eventSource string) string {
+func name(obj, eventType, eventSource, eventExtensionName, eventExtensionValue string) string {
 	// Pod names need to be lowercase. We might have an eventType as Any, that is why we lowercase them.
 	if eventType == "" {
 		eventType = "testany"
 	}
 	if eventSource == "" {
 		eventSource = "testany"
 	}
-	return strings.ToLower(fmt.Sprintf("%s-%s-%s", obj, eventType, eventSource))
+	if eventExtensionName == nilString {
```

Comment:
will have to be with a for loop. Maybe add a TODO that we should be careful with the size, as name sizes cannot exceed 253 chars I think